dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -247,6 +225,39 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
         return tool;
     }
 
+    /**
+     * Get an openapi descriptor type from a descriptor language file type
+     *
+     * @param fileType a descriptor language file type such as FileType.DOCKSTORE_CWL
+     * @return an Optional openapi descriptor type such as GALAXY("GALAXY") or SMK("SMK");
+     */
+    public static Optional<DescriptorType> getDescriptorTypeFromFileType(FileType fileType) {
+        DescriptorLanguage descriptorLanguage;
+        try {
+            descriptorLanguage = DescriptorLanguage.getDescriptorLanguage(fileType);
+        } catch (Exception e) {
+            return Optional.empty();
+        }
+
+        return getDescriptorTypeFromDescriptorLanguage(descriptorLanguage);
+    }
+
+    /**
+     * Get an openapi descriptor type from a descriptor language
+     *
+     * @param descriptorLanguage a descriptor language such as DescriptorLanguage.GXFORMAT2 or DescriptorLanguage.SMK
+     * @return an Optional openapi descriptor type such as GALAXY("GALAXY") or SMK("SMK");
+     */
+    public static Optional<DescriptorType> getDescriptorTypeFromDescriptorLanguage(DescriptorLanguage descriptorLanguage) {
```

Comment:
You could just put a `getDescriptorType` method in `DescriptorLanguage`. Then you don't need this method, and the existing caller of this method just invokes `descriptorLanguage.getDescriptorType()`.

On the one hand, it feels like the info belongs there; on the other hand, the DescriptorLanguage constuctor is getting pretty big.

Reply:
I don't think that would work because DescriptorLanguage doesn't import package io.openapi.model DescriptorType, and maybe should not import that package?

Reply:
Ah, good point.

Reply:
If we implemented the `matchesShortString` method mentioned above, we might be able to use it to simplify this method, too.

Reply:
Made a ticket for your suggestion  https://ucsc-cgl.atlassian.net/browse/SEAB-4485